Codeinwp/templates-patterns-collection
# Feat: improve Templates Cloud to better showcase as an independent plugin

### Summary

Moved "My Library" to a new location under the "Templates Cloud" menu. (pic. 1)
Introduced a new page "Settings" under the "Templates Cloud" menu. This page now holds the activation for the plugin and the feedback form. (pic. 1)
CSS style changes in line with the new Design from Figma. (pic. 3)
Fixed issue with the `<Popover />` component on the new versions of WordPress
Fixed E2E tests

The new menu items ( "My Library" and "Settings" ) are displayed for all users with `manage_options` capability since Saving to the Library is also available for these users.

"Starter Sites" and "Page Templates" remain with  `plugin_activation` capability as they will require the user to be able to install and activate plugins/themes if the Starter Site or Template requires them.

### Screenshots

<details open>
    <summary>New Menu layout w. Neve (pic. 1)</summary>

![image](https://github.com/Codeinwp/templates-patterns-collection/assets/23024731/fe5aacd1-c15a-4faf-b0e2-a110ad2218ce) 
![image](https://github.com/Codeinwp/templates-patterns-collection/assets/23024731/2a98d4cf-f821-462c-af4f-52ebd56e51cf)
</details>

<details>
    <summary>New Menu layout w./o. Neve (pic. 2)</summary>

![image](https://github.com/Codeinwp/templates-patterns-collection/assets/23024731/c798f346-0a27-4a95-88bc-52f08c180441)
</details>

<details>
    <summary>Settings page (pic. 3)</summary>

![image](https://github.com/Codeinwp/templates-patterns-collection/assets/23024731/3fd64a47-703a-4451-a5bc-2afda3ed9178)
![image](https://github.com/Codeinwp/templates-patterns-collection/assets/23024731/1fdb97aa-4bc8-454a-8dff-d67f3c0c1336)

</details>

### Test instructions


**I. Test using Neve Free**
1. On a fresh instance with Neve, add the Templates Patterns Collection plugin
2. Check the new menu locations (pic. 1)
3. Check that "My Library" is not accessible if no License is provided and that it will redirect to the License activation page from "Settings" when the "I already have a key" link is clicked.
4. Activate using a Neve Pro Key or a Templates Cloud Key
5. Check that the Library is available
6. Check that on a new Post you can Import a template from the Library using the Templates Cloud Block
7. Check the Feedback Form from "Settings"
8. Check the links present on the "Settings" page. (Useful Links)
9. Check the Starter Sites and Page Templates work as before

**II. Test using Neve Pro**
1. On a fresh instance with Neve and Neve Pro where Neve PRO is activated, install the Templates Patterns Collection plugin
2. Check that the Starter Sites and Page Templates work as before
3. Check that "My Library" is already accessible and works as before from the new location.
4. Check that on a new Post you can Import a template from the Library using the Templates Cloud Block
5. Check the White Label feature interacts and works as before with the Templates Patterns Collection plugin

**III. Test using a different theme, other than Neve**
1. On a fresh instance without Neve, add the Templates Patterns Collection plugin
2. Check all the steps from **test case I** except point 9. which does not apply here.
3. Instead of point 9 check that Starter Sites and Page Templates don't appear in the Appearance menu.


Closes codeinwp/templates-cloud#74.
